src/commonMain/kotlin/com/ashampoo/kim/format/jpeg/JpegRewriter.kt

```diff
@@ -156,60 +113,21 @@ object JpegRewriter : BinaryFileParser() {
         byteWriter.write(JpegConstants.SOI)
 
         if (newBytes != null) {
-
-            val markerBytes = JpegConstants.JPEG_APP1_MARKER.toShort().toBytes(byteOrder)
-
             if (newBytes.size > JpegConstants.MAX_SEGMENT_SIZE)
                 throw ExifOverflowException("APP1 Segment is too long: " + newBytes.size)
 
-            val markerLength = newBytes.size + 2
-            val markerLengthBytes = markerLength.toShort().toBytes(byteOrder)
-            var index = 0
-
-            val firstSegment = newSegments[index] as JFIFPieceSegment
-
-            if (firstSegment.marker == JpegConstants.JFIF_MARKER)
-                index = 1
+            val firstSegment = newSegments.first()
+            val exifSegment = UnknownSegment(JpegConstants.JPEG_APP1_MARKER, newBytes)
 
-            val exifSegment =
-                JFIFPieceSegmentExif(JpegConstants.JPEG_APP1_MARKER, markerBytes, markerLengthBytes, newBytes)
-
-            newSegments.add(index, exifSegment)
+            if (firstSegment.marker == JpegConstants.JFIF_MARKER) {
+                newSegments.add(1, exifSegment)
+            } else {
+                newSegments.add(0, exifSegment)
+            }
         }
 
-        var app1Written = false
-
-        @Suppress("LoopWithTooManyJumpStatements")
         for (piece in newSegments) {
-
-            if (piece is JFIFPieceSegmentExif) {
-
-                /* Only replace first APP1 segment; skips others. */
```

Comment:
I am not completely certain whether this logic can be safely removed. I assumed there must be a reason for including it as it is from Commons Imaging. However, not everything in that library is documented, and I have retained such elements considering the possibility of their historical significance. Do you have any insights or knowledge on whether this logic is indeed unnecessary?

Reply:
Simply, `JFIFPieceSegmentExif` was *only* created in one place, a few lines above. It doesn't escape this function at all. It was simply added to the list at index `0` or `1`, and that list would get looped over immediately afterwards, never escaping the function. The writing code also did exactly what `UnknownSegment.write` does. You can even compare the code between this snippet and `UnknownSegment.write` to see that it was identical, save for a few exceptions thrown and null checks that were moved or they were already being checked anyways.
Also, since `JFIFPieceSegmentExif` was only created above, and it was hence only added *once* to the list, it only could exist 0 or 1 times in the list, and so the `app1Written` was useless.
My guess is that Commons obviously had a different encoding scheme, and so they didn't encode the difference between `JFIFPieceSegmentExif` and a regular `JFIFPieceSegment`, opting instead to doing a check on the marker, and hence there could've theoretically been multiple `app1` segments in their code, but in this version, there probably cannot be.